src/commands/editor.rs

```diff
@@ -0,0 +1,257 @@
+        stdout().flush().ok();
+
+        editor.metadata_root.tool_info.tool = TOOL_NAME.to_string();
+        editor.metadata_root.tool_info.version = TOOL_VERSION.to_string();
+
+        writeln!(writer, "{}", serde_json::to_string_pretty(&editor.metadata_root)?)?;
+
+        println!("Done.");
+
+        writer.flush()?;
+        
+        Ok(())
+    }
+}
+
+
+
+impl EditConfig {
+    pub fn from_path<P: AsRef<Path>>(path: P) -> Result<Self> {
+        let json_file = File::open(path)?;
+        let mut config: EditConfig = serde_json::from_reader(&json_file)?;
+        
+        if let Some(to_duplicate) = config.duplicate.as_mut() {
+            to_duplicate.sort_by_key(|meta| meta.offset);
+            to_duplicate.reverse();
+        }
+
+        Ok(config)
+    }
+
+    fn execute(self, metadata: &mut MetadataJsonRoot) -> Result<()> {
+        let mut do_renumber: bool = false;
+
+        // Drop metadata frames
+        if let Some(ranges) = &self.remove {
+            self.remove_frames(ranges, &mut metadata.scene_info)?;
+
+            do_renumber = true;
+        }
+
+
+        if let Some(to_duplicate) = &self.duplicate {
+            self.duplicate_metadata(to_duplicate, &mut metadata.scene_info)?;
+
+            do_renumber = true;
+        }
+
+        let scene_cuts: HashSet<usize> = self.scene_cuts.unwrap_or(Vec::new()).into_iter().collect();
+
+        if !scene_cuts.is_empty() || do_renumber {
+            let mut new_scene_first_frame_index: Vec<usize> = Vec::new();
+            let mut new_scene_frame_numbers: Vec<usize> = Vec::new();
+
+            let mut curr_scene_id: usize = metadata.scene_info[0].scene_id;
+            let mut new_scene_id: usize = 0;
+            let mut new_scene_frame_index: usize = 0;
+
+            new_scene_first_frame_index.push(0);
+
+            for idx in 0..metadata.scene_info.len(){
+                metadata.scene_info[idx].sequence_frame_index = idx;
+
+                if curr_scene_id != metadata.scene_info[idx].scene_id || scene_cuts.contains(&idx){
+                    if curr_scene_id != metadata.scene_info[idx].scene_id {
+                        curr_scene_id = metadata.scene_info[idx].scene_id;
+                    }
+
+                    new_scene_first_frame_index.push(idx);
+                    new_scene_frame_numbers.push(new_scene_frame_index);
+
+                    new_scene_id += 1;
+                    new_scene_frame_index = 0;
+                }
+                metadata.scene_info[idx].scene_id = new_scene_id;
+                metadata.scene_info[idx].scene_frame_index = new_scene_frame_index;
+                new_scene_frame_index += 1;
+            }
+            new_scene_frame_numbers.push(new_scene_frame_index);
+
+
+            metadata.scene_info_summary.scene_first_frame_index = new_scene_first_frame_index;
+            metadata.scene_info_summary.scene_frame_numbers = new_scene_frame_numbers;

```

Comment:
All of this renumbering logic should not be necessary.
When using `generate_json` on the final `Vec` of metadata, the scene info will be computed correctly.